File: 251003/Dedov/Lab4-5/publisher/src/main/java/by/bsuir/publisher/dto/requests/NewsRequestDto.java
package by.bsuir.publisher.dto.requests;

import by.bsuir.publisher.dto.requests.BaseRequestDto;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@NoArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
public class NewsRequestDto extends BaseRequestDto {
    private Long editorId;

    @Size(min = 3, max = 32)
    private String title;

    @Size(min = 3, max = 32)
    @Pattern(regexp="^.*[a-zA-Z]+.*$")
    private String content;
}

File: 251001/Lashkin/publisher/DTO/RequestDTO/NewsRequestDTO.cs
﻿namespace Publisher.DTO.RequestDTO;

public class NewsRequestDTO
{
    public long Id { get; set; }
    public string Title { get; set; }
    
    public long UserId { get; set; }

    public string Content { get; set; }
    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }
}